sdk/identity/identity/src/errors.ts

```diff
@@ -182,3 +184,31 @@ function convertOAuthErrorResponseToErrorResponse(errorBody: OAuthErrorResponse)
     traceId: errorBody.trace_id
   };
 }
+
+/**
+ * Error used to enforce authentication after trying to retrieve a token silently.
+ */
+export class AuthenticationRequiredError extends Error {
+  /**
+   * The options used to configure the getToken request.
+   */
+  public getTokenOptions?: GetTokenOptions;
+
+  constructor(
+    /**
+     * The list of scopes for which the token will have access.
+     */
+    public scopes: string[],
+    /**
+     * Optional parameters. A message can be specified. The {@link GetTokenOptions} of the request can also be specified to more easily associate the error with the received parameters.
+     */
+    options: {
```

Comment:
Seems like the options themselves aren't optional. Does `new AuthenticationRequiredError(["openid", "profile"])` compile?

I would create a type name for this options bag and declare it as `options: AuthenticationRequiredErrorOptions = {}`.

Reply:
ohh you’re correct. I’m a bit sleep deprived today

Reply:
I thought I was being minimalistic (but I was just lazy)